include/fixed_containers/tuples.hpp

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include "fixed_containers/tuples_as_tuple_view.hpp"
+
+#include <concepts>
+#include <cstddef>
+#include <tuple>
+#include <utility>
+
+namespace fixed_containers::tuples
+{
+template <std::size_t FIELD_COUNT, typename T>
+constexpr auto as_tuple_view(T& data)
+{
+    return as_tuple_view_detail::as_tuple_view<FIELD_COUNT, T>(data);
+}
+
+template <typename Tuple, typename Func>
+    requires(std::tuple_size_v<std::decay_t<Tuple>> == 0)
+constexpr void for_each_entry(Tuple&& /*tuple*/, Func&& /*func*/)
+{
+}
+
+template <typename Tuple, typename Func>
+    requires(
+        std::tuple_size_v<std::decay_t<Tuple>> > 0 and
+        std::invocable<Func,
+                       std::add_lvalue_reference_t<std::tuple_element_t<0, std::decay_t<Tuple>>>>)
```

Comment:
Is this just checking for the function being able to take the first element of a tuple as an argument? We should at least note that that isn't a comprehensive test

Reply:
This is to differentiate the 2 overloads of `for_each_entry()`.
This allows you to have a lambda that just gives a field reference or a lambda that has index + field reference.